llvm-spirv/test/extensions/INTEL/SPV_INTEL_cache_controls/multiple-decoration-single-arg.ll

```diff
@@ -3,8 +3,8 @@
 ; RUN: llvm-spirv --spirv-ext=+SPV_INTEL_cache_controls %t.bc -o %t.spv
 ; RUN: llvm-spirv -r %t.spv --spirv-target-env=SPV-IR -o - | llvm-dis -o - | FileCheck %s --check-prefix=CHECK-LLVM
 
-; CHECK-SPIRV-DAG: Name [[#Func:]] "test"
-; CHECK-SPIRV-DAG: Name [[#FuncGEP:]] "test_gep"
+; CHECK-SPIRV-DAG: EntryPoint [[#]] [[#Func:]] "test"
```

Comment:
Tagging @MrSidims here, because this looks like a plain bug in the test, i.e. this change should be upstreamed later.
Both `test` and `test_gep` are kernels and therefore should be represented as `EntryPoint` instruction in SPIR-V. There could be an [`OpName`](https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpName) instruction referencing them as well, but it is a part of debug info and it can be safely removed from the module. It is just a coincidence that the test works using it.

I assume that the same change will be applied in the upstream translator and therefore it looks ok to me to proceed with it at intel/llvm

Reply:
It's not a bug in the test, it's a missing patch in intel/llvm (AFAIK it's being restored, it was missing due to lack of support in IGC).
Basically the missing patch was doing to following: for every kernel function it is creating a wrapper entry point that calls the function with the code copied form the original kernel. It was done because OpenCL allows one kernel to call another kernel and to comply the requirement from SPIR-V spec that one EntryPoint must not call another EntryPoint, we had to create such wrapper (actually, translator's community has done it for us).